nadlogar/problems/models.py

```diff
@@ -239,3 +239,149 @@ def generate(self):
         razstavljen = sympy.factor(izraz)
 
         return {"izraz": sympy.latex(izraz), "razstavljen": sympy.latex(razstavljen)}
+
+
+class ElementiMnozice(Problem):
+    """Problem za izpis elementov množice iz podanega predpisa"""
+
+    linearna_kombinacija = models.BooleanField(
+        "linearna kombinacija",
+        help_text="Ali naj naloga vsebuje linearno kombinacijo?",
+        choices=[(True, "Da"), (False, "Ne")],
+    )
+
+    class Meta:
+        verbose_name = "elementi množice"
+
+    def generate(self):
+        pogoj = random.choice(["|", "<", "<="])
+        n = sympy.symbols("n")
+        if not self.linearna_kombinacija:
+            a = 1
+            b = 0
+        else:
+            a = random.randint(1, 3)
+            b = random.choice([-2, -1, 0, 1, 2])
+        if pogoj == "|":
+            stevilo = random.randint(15, 45)
+            ustrezni = sympy.divisors(stevilo)
+        elif pogoj == "<":
+            stevilo = random.randint(5, 12)
+            ustrezni = list(range(1, stevilo))
+        elif pogoj == "<=":
+            stevilo = random.randint(5, 8)
+            ustrezni = list(range(1, stevilo + 1))
+        mnozica = sympy.FiniteSet(*[a * x + b for x in ustrezni if a * x + b > 0])
+        pogoj_latex = {"|": r"\mid", "<": r"\lt", "<=": r"\le"}
+        return {
+            "n": sympy.latex(sympy.simplify(a * n + b)),
+            "pogoj": pogoj_latex[pogoj],
+            "stevilo": sympy.latex(stevilo),
+            "mnozica": sympy.latex(mnozica),
+        }
+
+
+class PotencnaMnozica(Problem):
+    """Problem za izpis potenčne množice od dane množice"""
+
+    class Meta:
+        verbose_name = "potenčna množica"
+
+    def generate(self):
+        velikost = random.randint(2, 3)
+        mnozice = [
```

Comment:
Za konstatne spremenljivke se ponavadi uporabi `tuple`, razen če je list potreben.

Reply:
S tem se ne bi čisto strinjal. Za zbirko homogenih podatkov poljubne dolžine (tudi, če se ta ne bo spreminjala) se uporablja sezname, nabore pa se uporablja za potenicalno heterogen kartezični produkt. To odražajo tudi tipi, ki so `list[X]` ter `tuple[X, Y, Z, ...]`.

Reply:
A res, meni se zdi da to ni toliko stvar tipa, kot je problem (im)mutabilitya. Če je nekaj konstanta, in jo lahko naredis immutable, zakaj bi jo pustil mutable? Je bolj varno in manj potratno.